src/ja/comicgrast/src/eu/kanade/tachiyomi/extension/ja/comicgrast/ComicGrast.kt

```diff
@@ -0,0 +1,109 @@
+package eu.kanade.tachiyomi.extension.ja.comicgrast
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.parseAs
+import keiyoushi.utils.tryParse
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class ComicGrast : HttpSource() {
+    override val name = "Comic Grast"
+    override val baseUrl = "https://novema.jp"
+    override val lang = "ja"
+    override val supportsLatest = false
+
+    private val dateFormat = SimpleDateFormat("yyyy/MM/dd", Locale.JAPAN)
+
+    override val client = network.cloudflareClient.newBuilder()
+        .addInterceptor(ImageInterceptor())
+        .build()
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/comic/serial/$page", headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.select("ul.comicList > li").map { element ->
+            SManga.create().apply {
+                title = element.select("dl > dt > p.line-clamp.n2").text()
+                setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+                thumbnail_url = element.selectFirst("img")?.absUrl("data-src")
+            }
+        }
+        val hasNextPage = document.selectFirst(".pageList li.next") != null
+        return MangasPage(mangas, hasNextPage)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/comic/search/$page".toHttpUrl().newBuilder()
+            .addQueryParameter("type", "0")
+            .addQueryParameter("word", query)
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        return popularMangaParse(response)
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val document = response.asJsoup()
+        return SManga.create().apply {
+            title = document.selectFirst(".comicTit")?.text()?.trim().toString()
+            description = document.selectFirst(".comicStory .txtAcd_text_inner")?.text()?.trim()
```

Comment:
`.text()` is already trimmed according to Jsoup docs.
```suggestion
            description = document.selectFirst(".comicStory .txtAcd_text_inner")?.text()
```